android-core/src/androidTest/kotlin/com.mparticle/networking/NetworkOptionsTest.kt

```diff
@@ -181,6 +188,161 @@ class NetworkOptionsTest : BaseCleanInstallEachTest() {
+        Assert.assertEquals(
+            audienceUrl,
+            identityClient.getUrl(MParticleBaseClientImpl.Endpoint.AUDIENCE).authority
+        )
+        Assert.assertEquals(
+            configUrl,
+            identityClient.getUrl(MParticleBaseClientImpl.Endpoint.CONFIG).authority
+        )
+        Assert.assertEquals(
+            eventsUrl,
+            identityClient.getUrl(MParticleBaseClientImpl.Endpoint.EVENTS).authority
+        )
+        Assert.assertEquals(
+            identityUrl,
+            identityClient.getUrl(MParticleBaseClientImpl.Endpoint.IDENTITY).authority
+        )
+        randIdentityPath = mRandomUtils.getAlphaString(10)
+        Assert.assertEquals(
+            "/v1/$randIdentityPath",
+            identityClient.getUrl(
+                MParticleBaseClientImpl.Endpoint.IDENTITY,
+                randIdentityPath
+            ).path
+        )
+
+        // test the that the Path is still the default one (make sure the overrideSubdirectory is not kicking in when it shouldn't)
+        Assert.assertEquals(
+            defaultUrls[MParticleBaseClientImpl.Endpoint.AUDIENCE]?.path,
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.AUDIENCE).path
+        )
+        Assert.assertEquals(
+            defaultUrls[MParticleBaseClientImpl.Endpoint.CONFIG]?.path,
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.CONFIG).path
+        )
+        Assert.assertEquals(
+            defaultUrls[MParticleBaseClientImpl.Endpoint.EVENTS]?.path,
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.EVENTS).path
+        )
+        Assert.assertEquals(
+            defaultUrls[MParticleBaseClientImpl.Endpoint.IDENTITY]?.path,
+            mpClient.getUrl(
+                MParticleBaseClientImpl.Endpoint.IDENTITY,
+                MParticleBaseClientImpl.Endpoint.IDENTITY.name
+            ).path
+        )
+    }
+
+    @Test
+    @Throws(MalformedURLException::class)
+    fun testRandomEndpoint_when_No_Domain_Mapping_And_DirectURL_Routing_false() {
+        val options = MParticleOptions.builder(mContext)
+            .credentials("us1-14a65s4d65a4ds", "jasjgasgauysdkabmnx")
+            .build()
+        MParticle.start(options)
+        setClients()
+        Assert.assertEquals(
+            NetworkOptionsManager.MP_URL_PREFIX.addSuffix(),
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.AUDIENCE).authority
+        )
+        Assert.assertEquals(
+            NetworkOptionsManager.MP_CONFIG_URL,
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.CONFIG).authority
+        )
+        Assert.assertEquals(
+            NetworkOptionsManager.MP_URL_PREFIX.addSuffix(),
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.EVENTS).authority
+        )
+        Assert.assertEquals(
+            NetworkOptionsManager.MP_IDENTITY_URL_PREFIX.addSuffix(),
+            mpClient.getUrl(MParticleBaseClientImpl.Endpoint.IDENTITY).authority
+        )
+    }
+
+    /*
```

Comment:
If this test can't run, should we just remove it? I don't like having big commented out blocks of code in the codebase... We can always pull it back out of git history if/when we're ready to add it back.

Reply:
Mansi - can you make a story ticket for this and add it under [this epice](https://mparticle-eng.atlassian.net/browse/SQDSDKS-6612)?  You could copy and paste the below code in there to easily bring it back in a future follow up.  I think the ticket should be to add the library you were talking about but said you were uncertain if you could add in order to mock the request.

Reply:
@rmi22186  created here SQDSDKS-6613